mobile-app/app/screens/AppNavigator/screens/Balances/hooks/TokenPrice.ts

```diff
@@ -0,0 +1,116 @@
+import { useEffect } from 'react'
+import BigNumber from 'bignumber.js'
+import { RootState } from '@store'
+import { fetchPoolPairs } from '@store/wallet'
+import { PoolPairData } from '@defichain/whale-api-client/dist/api/poolpairs'
+import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
+import { useDispatch, useSelector } from 'react-redux'
+import { checkIfPair, findPath, GraphProps } from '@screens/AppNavigator/screens/Dex/helpers/path-finding'
+
+interface CalculatePriceRatesI {
+  aToBPrice: BigNumber
+  bToAPrice: BigNumber
+  estimated: BigNumber
+}
+
+interface DexTokenPrice {
+  getTokenPrice: (symbol: string, amount: string, isLPS?: boolean) => BigNumber
+  calculatePriceRates: (fromTokenSymbol: string, pairs: PoolPairData[], amount: string) => CalculatePriceRatesI
+  getArbitraryPoolPair: (tokenASymbol: string, tokenBSymbol: string) => PoolPairData[]
+}
+
+export function useTokenPrice (): DexTokenPrice {
+  const client = useWhaleApiClient()
+  const dispatch = useDispatch()
+  const blockCount = useSelector((state: RootState) => state.block.count)
+  const pairs = useSelector((state: RootState) => state.wallet.poolpairs)
+
+  const graph: GraphProps[] = pairs.map(pair => {
+    return {
+      pairId: pair.data.id,
+      a: pair.data.tokenA.symbol,
+      b: pair.data.tokenB.symbol
+    }
+  })
+
+  useEffect(() => {
+    dispatch(fetchPoolPairs({ client }))
+  }, [blockCount])
+
+  function getTokenPrice (symbol: string, amount: string, isLPS: boolean = false): BigNumber {
+    if (isLPS) {
+      const pair = pairs.find(pair => pair.data.symbol === symbol)
+      if (pair === undefined) {
+        return new BigNumber('')
+      }
+      const ratioToTotal = new BigNumber(amount).div(pair.data.totalLiquidity.token)
+      const tokenAAmount = ratioToTotal.times(pair.data.tokenA.reserve).decimalPlaces(8, BigNumber.ROUND_DOWN)
+      const tokenBAmount = ratioToTotal.times(pair.data.tokenB.reserve).decimalPlaces(8, BigNumber.ROUND_DOWN)
+      const usdTokenA = getTokenPrice(pair.data.tokenA.symbol, tokenAAmount.toFixed(8))
+      const usdTokenB = getTokenPrice(pair.data.tokenB.symbol, tokenBAmount.toFixed(8))
+      return usdTokenA.plus(usdTokenB)
+    }
+    // active price for walletTokens based on USDT
+    const arbitraryPoolPair = getArbitraryPoolPair(symbol, 'USDT')
```

Comment:
if the symbol is `USDT`, it won't find an `arbitraryPoolPair` which will cause not to show any USD value
![image](https://user-images.githubusercontent.com/13292662/151082760-3fad63ef-3e49-4efc-9451-dc6e060e686d.png)

Reply:
This also happens to LPs with `USDT`
![image](https://user-images.githubusercontent.com/13292662/151083112-98e56d0e-9518-4d35-b583-8dc7298e34ef.png)

Reply:
Pushed a fix for this